api/errors/error.go

```diff
@@ -0,0 +1,25 @@
+package errors
+
+// This file contains all of the errors that can be generated from the
+// docker engine but are not tied to any specific top-level component.
+
+import (
+	"net/http"
+
+	"github.com/docker/distribution/registry/api/errcode"
+)
+
+const errGroup = "engine"
+
+var (
+	// ErrorCodeGeneric is generated when there is no better error message
+	// available. This will typically happen when some low-level function
+	// is called and we just need to bubble the unknown error back up to
+	// the API. We should try to avoid using this.
+	ErrorCodeGeneric = errcode.Register(errGroup, errcode.ErrorDescriptor{
```

Comment:
I see. `ErrorCodeEngineUnknown`?

I think part of the problem is that this use case is tying error codes too tightly to the UI. Frankly, I don't see what the value of redefining this from "unknown error" with a useful message to "An unexpected error occurred" (ignoring the fact that this error message violates [Go's style guide](https://github.com/golang/go/wiki/CodeReviewComments#error-strings)). The way the error code was structured, errors of this nature would appear as "unknown error: something unfortunate happened". Depending on how careful the error chaining is managed, usually the problem can be succinctly identified.